lexer.go

```diff
@@ -693,19 +695,19 @@ func (l *tomlLexer) lexNumber() tomlLexStateFn {
 				return l.errorf("float cannot end with a dot")
 			}
 			pointSeen = true
-		} else if next == 'e' || next == 'E' {
+		case next == 'e' || next == 'E':
 			expSeen = true
 			l.next()
 			r := l.peek()
 			if r == '+' || r == '-' {
 				l.next()
 			}
-		} else if isDigit(next) {
+		case isDigit(next):
 			digitSeen = true
 			l.next()
-		} else if next == '_' {
+		case next == '_':
 			l.next()
-		} else {
+		default:
```

Comment:
I think this changes the effect of the `break` statement below, which is supposed to break out of the loop, but not breaks out of the `switch`

Reply:
Yes you right. returned the previous code block